feat: add Contributors page

navaneethvenu opened the pull request:
- Add Contributors Page (A page displaying all the Contributors of the Community Project and their Contributions)

Todo:
  - Update Contributor, Contribution dummy data with real data
  - Update Theming
  - Use Better Hero Image (Display Dev Panel too)
  
Screenshots:
![Frame 32](https://github.com/dscmbcet/website-3.0/assets/69888461/d95ea9db-496e-4020-a45a-70649cea7d6e)

navaneethvenu committed:
feat: add base layout of contributor list sections

- initial Implementation of architects and builders grid-based sections and cards
- make data semi-dynamic (based on a data file)

TODO:
- fix colours
- add socials and profile pictures to data
- change time to be based on real timings

navaneethvenu committed:
chore: improve and fix page layout, add icons

- add github header in dev contributor section
- fix figma header in design contributor section
- add social icons in figma, github cards
- make padding grid based
- add linkedin, github logos

navaneethvenu committed:
chore: complete and refactor list section layouts

Layout:
- complete and refactor list section layout
- modify contributor datatypes and content
- make elapsed time on cards dynamic

General:
- add SVGR to use SVGs as react components
- add Phosphor Icons as dependency
- update Colour Variables

Todo:
- add real data for contributors and contributions

navaneethvenu committed:
chore: complete contributors page

- Add Hero Section

Todo:
- update contributors and contributions with real data
- update Hero Section Colours and Dark Theme images when theme is finalised

navaneethvenu committed:
Merge branch 'contributors-page' into dev

xditya reviewed (approved):
LGTM

aurora-0025 commented on src/app/about/mode-toggle.tsx:
@navaneethvenu 
This file is located in the 'about' folder. Considering its potential use in multiple pages, particularly within the navbar across various pages, it might be great to place it within the 'components' folder.

navaneethvenu replied:
I've moved everything to a sample folder as @RiyaMathew-11 suggested, the toggle is very different from what we'll be having, and hence this wont be reused. This toggle is only for reference, and will be deleted once the actual toggle arrives.

RiyaMathew-11 commented on src/app/contributors/components/figma-card.tsx:
```diff
@@ -0,0 +1,114 @@
+import Image from "next/image";
+import dateToElapsed from "@/utils/date-to-elapsed";
+import Contribution from "@/models/contributions";
+import BenGeorgeNetto from "@/../public/assets/images/team-photos/bengeorgenetto.png";
```
Let's use a good variable name instead of BenGeorgeNetto? - keep it standardised - like ProfileImage

RiyaMathew-11 commented on src/app/contributors/page.tsx:
```diff
@@ -0,0 +1,231 @@
+import GithubCard from "./components/github-card";
+import { contributions } from "@/data/contributors";
+import GitHubLogo from "@/../public/assets/icons/github-logo.svg";
+import CloudBottomImage from "@/../public/assets/images/CloudsBottom.png";
```
So usually when we name asset files - we keep the names in smallcase to avoid confusions in consistency. 

If we are enabling first-letter caps for file names, let's note to maintain this style for all asset files to make them consistent.

navaneethvenu replied:
fixed (kept names in kebab-case)

RiyaMathew-11 commented on src/app/about/page.tsx:
I feel - since this file is like a reference for font styles - keep this separate - maybe in a directory called app/sample/page.tsx - if this needs to be kept as a test page. 

As about page also comes here, this page would go when about page is pushed. Or if this isnt needed, we can remove this.

navaneethvenu replied:
Done, moved to app/sample

navaneethvenu committed:
chore: move example files to sample folder

- move styled components and theem toggle examples to a seperate example folder for reference

navaneethvenu committed:
fix: overflow issue for windows users

- add overflow-x-none to top level container on page.

navaneethvenu committed:
chore: update asset names to kebab case

navaneethvenu committed:
chore: standardize variable name for ProfileImage
